pkg/operator/controller/status/controller.go

```diff
@@ -341,6 +342,32 @@ func checkDNSAvailable(dns *operatorv1.DNS) bool {
 	return false
 }
 
+// computeOperatorUpgradeableCondition computes the operator's current
+// Upgradeable status state by examining the Upgradeable status of the DNS
+// operator
+func computeOperatorUpgradeableCondition(dns *operatorv1.DNS) configv1.ClusterOperatorStatusCondition {
+	var upgradeable bool
+	for _, cond := range dns.Status.Conditions {
+		if cond.Type == operatorv1.OperatorStatusTypeUpgradeable && cond.Status == operatorv1.ConditionTrue {
+			upgradeable = true
```

Comment:
Probably better to set `Upgradeable=True` if the DNS doesn't have an `Upgradeable` status condition:
```suggestion
	upgradeable := true
	for _, cond := range dns.Status.Conditions {
		if cond.Type == operatorv1.OperatorStatusTypeUpgradeable {
			upgradeable = cond.Status == operatorv1.ConditionTrue
```